sdcm/tester.py

```diff
@@ -834,22 +834,30 @@ def is_encrypt_keys_needed(self):
             'kmip_hosts' in append_scylla_yaml
         )
 
-    def prepare_kms_host(self) -> None:
-        version_supports_kms = (self.params.is_enterprise and
-                                ComparableScyllaVersion(self.params.scylla_version) >= '2023.1.3')
-        backend_support_kms = self.params.get('cluster_backend') in ('aws',)
-        kms_configured_in_sct = self.params.get('scylla_encryption_options')
-        test_uses_oracle = self.params.get("db_type") == "mixed_scylla"
-        should_enable_kms = (version_supports_kms and
-                             backend_support_kms and
-                             not kms_configured_in_sct and
-                             not test_uses_oracle and
-                             not self.params.get('enterprise_disable_kms'))
-
-        if should_enable_kms:
+    def prepare_kms_host(self) -> None:  # noqa: PLR0911
+        if self.params.get('cluster_backend') != 'aws':
+            logging.debug("Skip configuring AWS KMS, test is not running on AWS")
+            return
+        if self.params.get('enterprise_disable_kms'):
+            logging.debug("Skip configuring AWS KMS, `enterprise_disable_kms` is set in the config")
+            return
+        if self.params.get("db_type") == "mixed_scylla":
+            logging.debug("Skip configuring AWS KMS, test uses mixed scylla versions")
+            return
+        try:
+            scylla_version = ComparableScyllaVersion(self.params.scylla_version)
+            if not (self.params.is_enterprise and scylla_version >= '2023.1.3'):
+                logging.debug(f"Skip configuring AWS KMS, Scylla version {scylla_version} does not support KMS")
+                return
+        except ValueError as e:
+            InfoEvent(
+                message=f"Wanted to enable AWS KMS, but version check raised an error: {e}", severity=Severity.ERROR).publish()
+            return
```

Comment:
I think we should follow the `best effort` approach here.
Skipping the AWS KMS setting must be avoided if possible.

Options by priority:
- Remove version check as obsolete, see above comment.
- Try to get version elsewhere. SCT gets to know the Scylla version in other parts of the code.
- Ignore version check failure and try to install AWS KMS anyway. Still record an ERROR SCT event.
- Fail test right away explicitly saying that the AWS KMS setup failed on version check.